### ISSUE TITLE
Save automatically when refactoring

### ISSUE THREAD

**Issue by g-plane** [microsoft/vscode#47573]
- VSCode Version: 1.22.1
- OS Version: macOS

**It's a feature request, not a bug report.**

Just like the feature "Save automatically when Search & Replace", after refactoring the code (by pressing `F2` key or using context menu), the VS Code should save the file(s) automatically.

**Comment by msultani** [comment]
Hello! I'm giving this issue a shot and wanted to ask a clarifying question.

My understanding is that I am trying to make vscode automatically save the file after refactoring (`F2`). I believe I've found where the refactoring takes place: `SelectRefactorCommand` class in `vscode/extensions/typescript-language-features/src/features/refactorProvider.ts`.

My question is regarding adding the code to save automatically once the refactoring takes place in this location. If anyone can help point me in the right direction or correct my understanding, I would greatly appreciate it. Thank you!

**Comment by TrevorKarjanis** [comment]
A temporary workaround is to use File > Save All (Ctrl+K S).

**Comment by vnenkpet** [comment]
Has this really been open for more than a year with no progress, or am I missing something here?

**Comment by dahu33** [comment]
PLEASE!

**Comment by astrowonk** [comment]
Yes, please put in a preference where refactoring or symbol renaming just does it and saves rather than open 5 files unsaved with the changes. If one uses source control, then it's easy to revert the changes anyway. I was bewildered when I used symbol renaming and 4 other files opened unsaved rather than just changing everything in place.

**Comment by b-zurg** [comment]
Seriously surprised this isn't a thing already.  This makes refactoring terrible cumbersome.

**Comment by astrowonk** [comment]
This feature request has been open for 2 years, are there any updates @jrieken @mjbvz?

**Comment by selrond** [comment]
What's the state of this?
Seems like the usecase is common enough. 
It's really suboptimal to see dozens of opened & unsaved files on every moderate refactor:

![image](https://user-images.githubusercontent.com/6603389/97574002-de723380-19ea-11eb-88b1-49e2cfe2da70.png)

**Comment by JoaquinScript** [comment]
Maybe prompt the users with the option to save all related files?

**Comment by stefanpl** [comment]
Honestly, I don't even see why having them unsaved would be desirable. So far, the rename feature works so flawlessly for me that I never needed to double-check anything. Having the changes unsaved, on the other hand, has very real drawbacks:
- compilation breaks until all files are saved
- if you use "save all files" as a workaround, you better make sure that you don't have pending changes in other files that you didn't want saved.
- I suddenly have a truckload of open editors. Either I close them manually, or all at once, thereby closing also the editors I had open before starting the refactor

IMO there should be a setting to change this behaviour, and it should be the default as well.

**Comment by djfm** [comment]
Yes that would be a nice feature, I'm constantly forgetting to save files after a rename and spend considerable amounts of time figuring out what broke.

**Comment by Broyojo** [comment]
this feature seems like such a large oversight, I mean this takes away the advantage of rename symbol: automation. Now you have to comb through each file and save them all of you have to massively clutter up your workspace.

**Comment by fis-cz** [comment]
Auto save should work only if another file was not modified previously. Otherwise it should be optional to show dialog and let user decide.

**Comment by dec0dOS** [comment]
Any updates?

**Comment by reilem** [comment]
> This looks to be resolved now for me. If you check off File -> Auto Save

But this will turn on auto-save for everything. If you have a file-listener turned on to restart a server every time a file is saved it will restart constantly before your changes can even compile. So not really a solution for some of us.

**Comment by JosephTLyons** [comment]
Sort of bewildered when I renamed a variable and saw nearly 300 with changes needing to be saved.  A bit more bewildered to see that this issue is almost 4 years old.

**Comment by rzhao271** [comment]
What are some verification steps for this issue? I set `files.autoSave` to `"off"` and `files.refactoring.autoSave` to `true` and wasn't able to get the file to save automatically.

**Comment by jrieken** [comment]
To verify

* ensure `files.refactoring.autoSave` is one
* perform a cross-file refactoring, eg rename an exported typescript symbol
* make sure all files are saved

think about the single file case

**Comment by gtbono** [comment]
I just tried files.refactoring.autoSave in the new April release and it worked when I did for example a class rename using F2.

But when I moved a file to another folder by dragging it through the left sidebar. I told it to auto update the imports automatically and it did, but the files changed were opened and left unsaved.

Should refactoring autoSave work when updating imports on file move? I believe it should
